scripts/makeLegacyPlots.py

```diff
@@ -0,0 +1,1053 @@
+
+    afX       = array ("d", fX      )
+    afY       = array ("d", fY      )
+    afeYUp    = array ("d", feYUp   )
+    afeYDown  = array ("d", feYDown )
+    afeXRight = array ("d", feXRight)
+    afeXLeft  = array ("d", feXLeft )
+    gRatio = TGraphAsymmErrors (len(afX), afX, afY, afeXLeft, afeXRight, afeYDown, afeYUp);
+    
+    gRatio.SetMarkerStyle(8);
+    gRatio.SetMarkerSize(1.);
+    gRatio.SetMarkerColor(kBlack);
+    gRatio.SetLineColor(kBlack);
+    gRatio.SetName(newName)
+
+    return gRatio;
+
+## find maximum of tgraph, including error
+def findMaxOfGraph (graph):
+    uppers = []
+    for i in range (0, graph.GetN()):
+        x = Double(0.0)
+        y = Double(0.0)
+        graph.GetPoint(i, x, y)
+        uppers.append (y + graph.GetErrorYhigh(i))
+    return max(uppers)
+
+## remove negative bins and reset yield accordingly
+## NB: must be done BEFORE bin width division
+def makeNonNegativeHistos (hList):
+    for h in hList:
+        integral = h.Integral()
+        for b in range (1, h.GetNbinsX()+1):
+            if (h.GetBinContent(b) < 0):
+               h.SetBinContent (b, 0)
+        integralNew = h.Integral()        
+        if (integralNew != integral):
+            print "** INFO: removed neg bins from histo " , h.GetName() 
+        
+        # print h.GetName() , integral , integralNew
+        if integralNew == 0:
+            h.Scale(0)
+        else:
+            h.Scale(integral/integralNew) 
+
+def Xaxis2binNumber (histo):
+    Nbins = histo.GetNbinsX()
+
+    new_histo = TH1F(histo.GetName(), histo.GetName(), Nbins, 0.5, Nbins+0.5)    
+    for j in range(1,Nbins+1):
+        new_histo.SetBinContent(j,histo.GetBinContent(j))
+        new_histo.SetBinError(j,histo.GetBinError(j))
+
+    return new_histo
+    
+def makeStatSystUncertaintyBand (bkgSum,hBkgs,hBkgNameList,systCfgList,channel,selection,hShapes,hShapesNameList):
+    # we read the systematic configs to calculate the lnN systematic uncertainty
+    # we read all the shape variated historams to calculate the shape systematic uncertainty
+    # as an approximation we jut sum in quadrature all the sources and do not consider any correlation
+
+    nPoints   = bkgSum.GetNbinsX()
+    fX        = []
+    fY        = []
+    feYUp     = []
+    feYDown   = []
+    feXRight  = []
+    feXLeft   = []
+    syst_df   = pd.DataFrame(0.0,columns=hBkgNameList, index=['lnNup', 'lnNdown'])
+
+    # this for loop calculates the lnN uncertainty from the syst.cfg files
+    # NB store the squared % value of the source!!
+    for cfg in systCfgList:
+        # read the file
+        for line in open('config/'+cfg+'.cfg','r'):
+            # split lines and skip not useful ones
+            f = line.split()
+            if len(f) < 1: continue
+            if f[0].startswith('#'): continue # skip comments
+            if f[0].startswith('qq'): continue # skipp VBFHH systs
+            if f[0].startswith('gg'): continue # skip GGHH systs
```

Comment:
Here I am reading the systematics_<...>.cfg and the lines that start with qq/gg are the systematics associated to the VBFHH/GGHH samples. So I am effectively skipping the systematic source, which also means I am skipping the sample. I wrote 'skip VBFHH/GGHH systs' because of the file I am reading the lines from.

Reply:
ok it was just for my understanding!it's fine!